### PR TITLE
feat: add `kili.tags` and `kili.tag_project` methods

### PR DESCRIPTION
**Pull request by Jonas1312**
for kili.tag_project, I accept both tag ids and tag labels in the `tags` parameter. This assumes that nobody would create a tag label named after a tag id...

Not sure if we should accept tag ids at all? If yes, maybe add a second argument `tag_ids`?

wdyt?

### CODE REVIEW

File: src/kili/presentation/client/tag.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+from kili.utils.logcontext import for_all_methods, log_call
+from kili.utils.tqdm import tqdm
+
+
+@for_all_methods(log_call, exclude=["__init__"])
+class TagClientMethods:
+    """Methods attached to the Kili client, to run actions on tags."""
+
+    kili_api_gateway: KiliAPIGateway
+    http_client: requests.Session
```
Comment (PierreLeveau) [inline src/kili/presentation/client/tag.py:18]: not used here

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

File: src/kili/gateways/kili_api_gateway/tag/operations.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+"""GraphQL Tags operations."""
+
+GQL_CHECK_TAG = """
```
[the remark this answers is not in the document]
Comment (Jonas1312) [reply src/kili/gateways/kili_api_gateway/tag/operations.py:3]: I put all queries in the operations.py. So that we can reuse them in the tests

**Comment by theodu** [comment]
waiting for the rebase to review

**Comment by Jonas1312** [comment]
I'm still not sure if we should accept tag ids or tag labels as parameters of the tag methods

File: src/kili/use_cases/issue/__init__.py [hunk 3]
```diff
@@ -4,17 +4,20 @@
...
     """Issue use cases."""
 
-    def __init__(self, kili_api_gateway: KiliAPIGateway):
+    def __init__(self, kili_api_gateway: KiliAPIGateway) -> None:
         self._kili_api_gateway = kili_api_gateway
 
-    def create_issues(self, project_id, issues: List[IssueToCreateUseCaseInput]):
+    def create_issues(
+        self, project_id: str, issues: List[IssueToCreateUseCaseInput]
+    ) -> List[IssueId]:
```
Comment (theodu) [inline src/kili/use_cases/issue/__init__.py:20]: It is better to force the return type hint even if it is well infered from the function implem with inputs ?

**Comment by theodu** [comment]
> I'm still not sure if we should accept tag ids or tag labels as parameters of the tag methods

good question, I think that is very good to be able to provide both (as you did). however, I wonder what is the best between having one field to provide both or to separate them into two fields for the user.

**Comment by Jonas1312** [comment]
> > I'm still not sure if we should accept tag ids or tag labels as parameters of the tag methods
> 
> good question, I think that is very good to be able to provide both (as you did). however, I wonder what is the best between having one field to provide both or to separate them into two fields for the user.

yes maybe two separate parameters would be safer, wdyt @PierreLeveau

File: tests/integration/use_cases/test_tag.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    # When
+    fetched_tags = TagUseCases(kili_api_gateway).get_tags_of_project(
+        project_id="fake_proj_id", fields=["id", "label"]
+    )
+
+    # Then
+    assert fetched_tags == tags
+
+
+def test_when_tagging_project_then_it_tags_the_project(kili_api_gateway: KiliAPIGateway):
```
Comment (theodu) [inline tests/integration/use_cases/test_tag.py:37]: we can add a test to assert that it raises an error when a tag is not found in the orga tags

File: src/kili/use_cases/issue/__init__.py (hunk 3 above)
Comment (Jonas1312) [reply src/kili/use_cases/issue/__init__.py:20]: Doesn't cost much to add it. I find that it helps, especially during refactoring

File: src/kili/adapters/kili_api_gateway/project/__init__.py [hunk 5]
```diff
@@ -5,15 +5,14 @@
...
 from kili.domain.project import ProjectId
 from kili.exceptions import NotFound
 
+from ..base import BaseOperationMixin
 
-class ProjectOperationMixin:
-    """Mixin extending Kili API Gateway class with Assets related operations."""
 
-    graphql_client: GraphQLClient
+class ProjectOperationMixin(BaseOperationMixin):
```
Comment (PierreLeveau) [inline src/kili/adapters/kili_api_gateway/project/__init__.py:14]: with Project related operations

**Review by PierreLeveau — APPROVED** [review]

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
add separate field for tag id and tag name

File: src/kili/adapters/kili_api_gateway/project/__init__.py [hunk 6]
```diff
@@ -5,15 +5,14 @@
...
+    """Mixin extending Kili API Gateway class with Assets related operations."""
```
Comment (PierreLeveau) [inline src/kili/adapters/kili_api_gateway/project/__init__.py:15]: with Projects related operations

**Pull request merged**